cpp/src/arrow/array/array_base.h

```diff
@@ -232,6 +232,14 @@ class ARROW_EXPORT Array {
   /// \return DeviceAllocationType
   DeviceAllocationType device_type() const { return data_->device_type(); }
 
+  /// \brief Return the statistics of this Array
+  ///
+  /// This just delegates to calling statistics on the underlying ArrayData
+  /// object which backs this Array.
+  ///
+  /// \return const ArrayStatistics&
+  std::shared_ptr<ArrayStatistics> statistics() const { return data_->statistics; }
```

Comment:
The return type should be `std::shared_ptr<ArrayStatistics>&` and we should probably add `const ArrayStatistics` to the `shared_ptr` so that callers can't mutate the statistics through the shared pointer.

Reply:
@kou

Reply:
I agree with you. I don't know why I missed `const` and `&` here...

Let's add them: GH-44590